### ISSUE TITLE
[CI] CcsCommonYamlTestSuiteIT classMethod failing

### ISSUE THREAD

**Issue by alex-spies** [elastic/elasticsearch#108949]
Looks like the test suite timed out:
```
May 23, 2024 4:28:10 AM sun.util.locale.provider.LocaleProviderAdapter <clinit>
...
Mei 23, 2024 4:43:03 AM com.carrotsearch.randomizedtesting.ThreadLeakControl checkThreadLeaks
SEVERE: 137 threads leaked from SUITE scope at org.elasticsearch.test.rest.yaml.CcsCommonYamlTestSuiteIT: 
```

**Build scan:**
https://gradle-enterprise.elastic.co/s/xagwkakb32s2u/tests/:qa:ccs-common-rest:yamlRestTest/org.elasticsearch.test.rest.yaml.CcsCommonYamlTestSuiteIT

**Reproduction line:**
```
null
```

**Applicable branches:**
8.14

**Reproduces locally?:**
No

**Failure history:**
[Failure dashboard for `org.elasticsearch.test.rest.yaml.CcsCommonYamlTestSuiteIT#classMethod`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('classMethod'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.test.rest.yaml.CcsCommonYamlTestSuiteIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure excerpt:**
```
java.lang.Exception: Suite timeout exceeded (>= 900000 msec).

  at __randomizedtesting.SeedInfo.seed([5B78EF87B604D791]:0)

```

**Comment by mayya-sharipova** [comment]
1 failure in the last 7 days.

**Comment by andreidan** [comment]
The reason this failed is that the remote cluster tripped an assertion 
```
{"@timestamp":"2024-05-23T11:33:38.630Z", "log.level":"ERROR", "message":"fatal error in thread [elasticsearch[remote_cluster-0][search_worker][T#46]], exiting", "ecs.version": "1.2.0","service.name":"ES_ECS","event.dataset":"elasticsearch.server","process.thread.name":"elasticsearch[remote_cluster-0][search_worker][T#46]","log.logger":"org.elasticsearch.bootstrap.ElasticsearchUncaughtExceptionHandler","elasticsearch.cluster.uuid":"tfFP_ouHQ_CxmSzdGHTRtA","elasticsearch.node.id":"ugdOImqDT7-xJUvTT5g1Cg","elasticsearch.node.name":"remote_cluster-0","elasticsearch.cluster.name":"remote_cluster","error.type":"java.lang.AssertionError","error.stack_trace":"java.lang.AssertionError\n\tat org.apache.lucene.core@9.10.0/org.apache.lucene.search.TopScoreDocCollector$SimpleTopScoreDocCollector$1.collect(TopScoreDocCollector.java:75)\n\tat org.apache.lucene.core@9.10.0/org.apache.lucene.search.Weight$DefaultBulkScorer.scoreRange(Weight.java:277)\n\tat org.apache.lucene.core@9.10.0/org.apache.lucene.search.Weight$DefaultBulkScorer.score(Weight.java:236)\n\tat org.elasticsearch.server@8.14.0-SNAPSHOT/org.elasticsearch.search.internal.CancellableBulkScorer.score(CancellableBulkScorer.java:45)\n\tat org.apache.lucene.core@9.10.0/org.apache.lucene.search.BulkScorer.score(BulkScorer.java:38)\n\tat org.elasticsearch.server@8.14.0-SNAPSHOT/org.elasticsearch.search.internal.ContextIndexSearcher.searchLeaf(ContextIndexSearcher.java:431)\n\tat org.elasticsearch.server@8.14.0-SNAPSHOT/org.elasticsearch.search.internal.ContextIndexSearcher.search(ContextIndexSearcher.java:360)\n\tat org.elasticsearch.server@8.14.0-SNAPSHOT/org.elasticsearch.search.internal.ContextIndexSearcher.lambda$search$4(ContextIndexSearcher.java:345)\n\tat org.apache.lucene.core@9.10.0/org.apache.lucene.search.TaskExecutor$TaskGroup.lambda$createTask$0(TaskExecutor.java:117)\n\tat java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)\n\tat org.elasticsearch.server@8.14.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)\n\tat 
```

This is the assertion that got tripped https://github.com/apache/lucene/blob/releases/lucene/9.10.0/lucene/core/src/java/org/apache/lucene/search/TopScoreDocCollector.java#L75

**Comment by andreidan** [comment]
The first test that encountered the failure is `Test create, merge, and search dot_product` in `41_knn_search_byte_quantized`
```
REPRODUCE WITH: ./gradlew ':qa:ccs-common-rest:yamlRestTest' --tests "org.elasticsearch.test.rest.yaml.CcsCommonYamlTestSuiteIT" -Dtests.method="test {p0=search.vectors/41_knn_search_byte_quantized/Test create, merge, and search dot_product}" -Dtests.seed=5B78EF87B604D791 -Dtests.locale=ms -Dtests.timezone=America/Dawson -Druntime.java=22
org.elasticsearch.test.rest.yaml.CcsCommonYamlTestSuiteIT > test {p0=search.vectors/41_knn_search_byte_quantized/Test create, merge, and search dot_product} FAILED
    java.lang.AssertionError: Failure at [search.vectors/41_knn_search_byte_quantized:526]: expected [2xx] status code but api [search] returned [500 Internal Server Error] [---
    error:
      root_cause:
      - type: "node_disconnected_exception"
        reason: "[remote_cluster-0][127.0.0.1:38009][indices:data/read/search] disconnected"
        stack_trace: "org.elasticsearch.transport.NodeDisconnectedException: [remote_cluster-0][127.0.0.1:38009][indices:data/read/search]\
          \ disconnected\n"
      type: "node_disconnected_exception"
      reason: "[remote_cluster-0][127.0.0.1:38009][indices:data/read/search] disconnected"
      stack_trace: "org.elasticsearch.transport.NodeDisconnectedException: [remote_cluster-0][127.0.0.1:38009][indices:data/read/search]\
        \ disconnected\n"
    status: 500
    ]
        at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:610)
        at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:553)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
        at org.elasticsearch.test.cluster.local.DefaultLocalElasticsearchCluster$1.evaluate(DefaultLocalElasticsearchCluster.java:47)
        at org.elasticsearch.test.cluster.local.DefaultLocalElasticsearchCluster$1.evaluate(DefaultLocalElasticsearchCluster.java:47)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
        at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
        at java.base/java.lang.Thread.run(Thread.java:1570)
        Caused by:
        java.lang.AssertionError: expected [2xx] status code but api [search] returned [500 Internal Server Error] [---
        error:
          root_cause:
          - type: "node_disconnected_exception"
            reason: "[remote_cluster-0][127.0.0.1:38009][indices:data/read/search] disconnected"
            stack_trace: "org.elasticsearch.transport.NodeDisconnectedException: [remote_cluster-0][127.0.0.1:38009][indices:data/read/search]\
              \ disconnected\n"
          type: "node_disconnected_exception"
          reason: "[remote_cluster-0][127.0.0.1:38009][indices:data/read/search] disconnected"
          stack_trace: "org.elasticsearch.transport.NodeDisconnectedException: [remote_cluster-0][127.0.0.1:38009][indices:data/read/search]\
            \ disconnected\n"
        status: 500
        ]
            at org.junit.Assert.fail(Assert.java:89)
            at org.elasticsearch.test.rest.yaml.section.DoSection.checkResponseException(DoSection.java:540)
            at org.elasticsearch.test.rest.yaml.section.DoSection.execute(DoSection.java:396)
            at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:590)
            ... 43 more
```

**Comment by andreidan** [comment]
@benwtrent as usual, you got this covered ❤ 

These failures should be fixed in Elasticsearch 8.15 by: https://github.com/apache/lucene/pull/13356
